Modules/Quotas/src/Quotas.Application/Identities/Queries/GetIdentity/GetIdentityResponse.cs

```diff
@@ -1,36 +1,40 @@
 ﻿using Backbone.Modules.Quotas.Application.DTOs;
 using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
 using Backbone.Modules.Quotas.Domain.Aggregates.Metrics;
+using Backbone.Modules.Quotas.Domain.Metrics;
 
 namespace Backbone.Modules.Quotas.Application.Identities.Queries.GetIdentity;
 public class GetIdentityResponse
 {
-    public GetIdentityResponse(string identityAddress, IEnumerable<TierQuota> tierQuotas, IEnumerable<IndividualQuota> individualQuotas, IEnumerable<Metric> metrics)
+    public string Address { get; set; }
+    public IEnumerable<QuotaDTO> Quotas { get; set; }
+
+    public static async Task<GetIdentityResponse> Create(MetricCalculatorFactory metricCalculatorFactory, string identityAddress, IEnumerable<TierQuota> tierQuotas, IEnumerable<IndividualQuota> individualQuotas, IEnumerable<Metric> metrics, CancellationToken cancellationToken)
     {
         var quotasList = new List<QuotaDTO>();
-        quotasList.AddRange(individualQuotas.Select(q =>
-            new QuotaDTO(
-                q.Id,
-                QuotaSource.Individual,
-                new MetricDTO(metrics.First(m => m.Key == q.MetricKey)),
-                q.Max,
-                q.Period.ToString()
-            )
-        ));
-        quotasList.AddRange(tierQuotas.Select(q =>
-            new QuotaDTO(
-                q.Id,
-                QuotaSource.Tier,
-                new MetricDTO(metrics.First(m => m.Key == q.MetricKey)),
-                q.Max,
-                q.Period.ToString()
-            )
-        ));
 
-        Address = identityAddress;
-        Quotas = quotasList;
+        var allQuotas = (individualQuotas as IEnumerable<Quota>).Union(tierQuotas);
+
+        foreach (var quota in allQuotas)
+        {
+            var usage = await CalculateUsage(metricCalculatorFactory, quota, identityAddress, cancellationToken);
```

Comment:
Oh, and I just noticed: you should run parallelize the calculation. Otherwise, for n Quotas the creation of the response takes n*dbResponseTIme.

Reply:
Okay, since EF Core doesn't allow parallel execution of multiple db queries, we should postpone this.